include/core/mir/mutex.h

```diff
@@ -92,6 +99,15 @@ class Mutex
         return MutexGuard<Guarded>{std::unique_lock{mutex}, value};
     }
 
+    /**
+     * Unlock an acquired mutex
+     *
+     * This is a convenience method for cases when the mutex must be dropped
+     * but cannot easily be made to leave scope.
+     */
+    static void drop(MutexGuard<Guarded> /*to_drop*/)
```

Comment:
The syntax for this seems a little verbose.

 Also, it's not clear to me what state the guard is left in after a move. Does it throw an error if you try to access it? Segfault?

Reply:
Drop no longer exists